pkg/tfbridge/x/tokens.go

```diff
@@ -565,11 +574,12 @@ func aliasOrRenameResource(p *b.ProviderInfo, tfToken string, hist *tokenHistory
 		hist.Past = append(hist.Past, alias[tokens.Type]{
 			Name:      hist.Current,
 			InCodegen: true,
+			Version:   currentVersion,
 		})
 	}
 	for _, a := range hist.Past {
 		legacy := a.Name
-		if a.InCodegen {
+		if a.InCodegen && (a.Version == currentVersion || a.Version == 0) {
```

Comment:
Got it, I wasn't sure about that

Reply:
As a though experiment, if we were concerned about backwards compatibility:

We still don't want to keep names from version 0 forever. One solution would be to record the version we will remove a name at, instead of the version when the name was added. Since `0` is never a valid version to remove a name, we can safely update `0` to one past the current major version. We would remain backwards compatible without a long term compromise in functionality.